cvat-sdk/cvat_sdk/core/proxies/tasks.py

```diff
@@ -119,6 +121,35 @@ def upload_data(
                 url, list(map(Path, resources)), pbar=pbar, **data
             )
 
+        if wait_for_completion:
+            if status_check_period is None:
+                status_check_period = self._client.config.status_check_period
+
+            self._client.logger.info("Awaiting for task %s creation...", self.id)
+            status: models.RqStatus = None
+            while status != models.RqStatusStateEnum.allowed_values[("value",)]["FINISHED"]:
+                sleep(status_check_period)
+                (status, response) = self.api.retrieve_status(self.id)
+
+                self._client.logger.info(
+                    "Task %s creation status=%s, message=%s",
+                    self.id,
+                    status.state.value,
+                    status.message,
+                )
+
+                if (
+                    status.state.value
+                    == models.RqStatusStateEnum.allowed_values[("value",)]["FAILED"]
+                ):
+                    raise exceptions.ApiException(
+                        status=status.state.value, reason=status.message, http_resp=response
+                    )
+
+                status = status.state.value
```

Comment:
It's pretty confusing to have `status` be both an `RqStatus` and a string at different points in the program... IMO, it would be better to replace this with `if status.state.value == models.RqStatusStateEnum.allowed_values[("value",)]["FINISHED"]: break` and remove the loop condition. The `status` variable declaration could also be removed.

Reply:
Ok, agree.